aaelf64/aaelf64.rst

```diff
@@ -1778,6 +1783,68 @@ The ``PAUTH`` and ``ENCD`` operators are defined in `PAUTHABIELF64`_.
   | 597        | \-         | R\_AARCH64\_AUTH\_TLSDESC\_ADD\_LO12   | G(ENCD(GTLSDESC(S)))                 | See `PAUTHABIELF64`_ |
   +------------+------------+----------------------------------------+--------------------------------------+----------------------+
 
+Relocations for Structure Protection Extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Structure Protection Extension defines a number of static
+relocations. The Structure Protection Extension is described in
+[STRUCTPROT_]. The Structure Protection Extension is in Alpha state.
+
+The structure protection relocations use the following additional operator:
+
+- ``FUNCINIT(S + A)`` The place is relocated at run-time with a ``R_AARCH64_IRELATIVE`` relocation with no referenced symbol and the value of S + A in the addend field.
+
+.. class:: structure-protection-instruction-relocations
+
+.. table:: Structure Protection Instruction Relocations
+
+  +------------+------------+----------------------------------------+--------------------------------------+----------------------+
+  | ELF64 Code | ELF32 Code | Name                                   | Operation                            | Comment              |
+  +============+============+========================================+======================================+======================+
+  |    316     | \-         | R\_AARCH64\_PATCHINST                  | S + A                                | See below            |
+  +------------+------------+----------------------------------------+--------------------------------------+----------------------+
+
+The referenced symbol for ``R_AARCH64_PATCHINST`` must either be
+undefined, or have section index ``SHN_ABS``. If the referenced symbol
+is undefined the relocation has no effect, otherwise write bits [31:0]
+of X at 4 byte-aligned place P. Check that 0 <= X < 2\ :sup:`32`.
+
+``R_AARCH64_PATCHINST`` may occur at the same offset as another
+relocation, for example when patching a branch and link instruction
+with its associated ``R_AARCH64_CALL26`` relocation. The object
+producer is responsible for ordering ``R_AARCH64_PATCHINST`` after all
+other non ``R_AARCH64_PATCHINST`` relocations at the same
+``r_offset``. The ``R_AARCH64_PATCHINST`` relocation always starts a
+new relocation composition sequence.
+
+The requirements for a static linker that supports
+``R_AARCH64_PATCHINST`` are limited to resolving the relocation. All
+other static linker processing of relocations such as `Call and jump
+relocations`_, `Program Linkage Table (PLT) Sequences and Usage
+Models`_ and `Relocation optimization`_ may ignore
+``R_AARCH64_PATCHINST``.
+
+The intended use case for ``R_AARCH64_PATCHINST`` to be used to
```

Comment:
Sorry for the last-minute fly-by nit, but I think this sentence is missing an `is`:

the intended <something> is to..

I think we can also remove `to be used to` so we'd go from:

`The intended use case for ``R_AARCH64_PATCHINST`` to be used to replace...` -> `The intended use case for ``R_AARCH64_PATCHINST`` is to replace...`.

Reply:
Thanks for the spot. I'll upload a new patch.